activerecord/test/cases/inheritance_test.rb

```diff
@@ -315,6 +315,13 @@ def test_new_without_storing_full_sti_class
     end
   end
 
+  def test_new_when_passing_class_to_type_when_store_attributes_are_present_on_sti_class
+    without_store_full_sti_class do
+      item = Company.new(type: Firm)
```

Comment:
This was deprecated in 4.2 and removed in 5.0 IIRC but I'd have to double check. We definitely used to handle it.